website/docs/concepts/html/events.mdx

```diff
@@ -223,43 +207,44 @@ impl Component for Comp {
             //highlight-next-line
             let input = target.and_then(|t| t.dyn_into::<HtmlInputElement>().ok());
 
-            input.map(|input| Msg::InputValue(input.value()))
-        });
-
-        let on_dangerous_change = link.callback(|e: Event| {
-            let target: EventTarget = e
-                .target()
-                .expect("Event should have a target when dispatched");
-            // You must KNOW target is a HtmlInputElement, otherwise
-            // the call to value would be Undefined Behaviour (UB).
-            //highlight-next-line
-            Msg::InputValue(target.unchecked_into::<HtmlInputElement>().value())
-        });
-
-        html! {
-            <>
-                <label for="cautious-input">
-                    { "My cautious input:" }
-                    <input onchange={on_cautious_change}
-                        id="cautious-input"
-                        type="text"
-                    />
-                </label>
-                <label for="dangerous-input">
-                    { "My dangerous input:" }
-                    <input onchange={on_dangerous_change}
-                        id="dangerous-input"
-                        type="text"
-                    />
-                </label>
-            </>
-        }
+            if let Some(input) = input {
+                input_value_handle.set(input.value());
+            }
+        })
+    };
+
+    let on_dangerous_change = Callback::from(move |e: Event| {
+        let target: EventTarget = e
+            .target()
+            .expect("Event should have a target when dispatched");
+        // You must KNOW target is a HtmlInputElement, otherwise
+        // the call to value would be Undefined Behaviour (UB).
+        //highlight-next-line
+        input_value_handle.set(target.unchecked_into::<HtmlInputElement>().value());
```

Comment:
Erring on the side of caution and using `dyn_into::<HtmlInputElement>().expect("")` would be my preference for docs.

Also: https://github.com/yewstack/yew/pull/2511/files#diff-0e97978a1c940b9f1a1688f631827e98643af024539ca448ce41baa54bc52415R307

Reply:
This is here to just highlight that it exists as it is mentioned in the text.
Also you can notice `dyn_into` is being used on line 208 already

Reply:
A comment stating, "We are sure that this is input element so we can convert it to the appropriate type without checking" would solve the issue